### DISCUSSION TITLE
Help with Shiny// Implementation of pandemic correction

### DISCUSSION THREAD

**Discussion by tinneuro** [United4Surveillance/signal-detection-tool#315] [General]
Hi everyone,

I am currently working on issue #312 and have finished the main part of the algorithm implementation but I am struggling with the shiny implementation. It should be easy but I can't get it to run. The branch is called  new_algos_with_pandemic_correction.
https://github.com/United4Surveillance/signal-detection-tool/tree/new_algos_with_pandemic_correction

What I want is shown in the inst/debugging/pandemic_correction.R in this branch. If you run the app in this file at the end you can see and understand what I want. 
Basically I want if someone selects one of the algorithms "Mean", "Timetrend", "Harmonic without timetrend", "Harmonic with timetrend" that a button pops up with pandemic correction (tick it or not tick it). If ticked then a dateinput should appear. Somehow the dateinput does not appear and I really do not understand why. Furthermore what I want is that if pandemic_correction is not selected then the date should be NULL. 
Maybe someone sees why it is not working? In my test script it is working and I dont know why in the main app it does not. Would really appreciate your help. :)
Thanks!

**Comment by AnjaMJ** [comment]
Regarding the checkbox input:
The checkbox keeps returning NULL, because you are missing an 'ns' in the conditional panel definition:

`shiny::conditionalPanel(condition =  paste0("output['", ns("algorithm_glm_text"), "'] == 'true'"),
checkboxInput(ns("pandemic_correction"), "Covid19 Pandemic Correction", value = FALSE))`

With the above edit the checkbox now returns values of TRUE and FALSE and your dateinput appears as it should :)

**Reply by tinneuro** [reply]
aaah wow so easy!! Thanks. I knew that I must have missed something easy 👍

**Comment by tinneuro** [comment]
If somebody has time and resources: on this branch in the plot of the timeseries somehow I get  gap exactly at the transition of the training to "test/outbreak detection" period. Happy to have someone investigating this.
